labman/gui/templates/library_prep_16S.html

```diff
@@ -9,27 +9,35 @@
 <script type='text/javascript'>
   function prepopulateGUI() {
     var processId = {% raw process_id if process_id else 'null' %};
-    var platesInfo = {% raw plates_info %};
+    var gdnaPlate = '{{gdna_plate}}';
+    var epmotion = '{{epmotion}}';
+    var epmotionTM300 = '{{epmotion_tm300}}';
+    var epmotionTM50 = '{{epmotion_tm50}}';
+    var masterMix = '{{master_mix}}';
+    var waterLot = '{{water_lot}}';
+    var primerPlate = '{{primer_plate}}';
+    var prepDate = '{{preparationDate}}';
+    var volume = '{{volume}}';
+
     if (processId !== null) {
       // We need to prepopulate the interface
       // Start by adding all plates:
-      var requests = [];
-      $.each(platesInfo, function(idx, pInfo) {
-        requests.push(addPlate(pInfo['Plate']));
-      });
+      var requests = [addPlate(gdnaPlate)];
       // We need to wait until all requests are done, so the interface will
       // be ready to be modified
       $.when.apply($, requests).then(function () {
-        $.each(platesInfo, function(idx, pInfo) {
-          var plateId = pInfo['Plate'];
-          $('#sel-primer-' + plateId).val(pInfo['Primer Plate']);
-          $('#epmotion-robot-' + plateId).val(pInfo['EpMotion']);
-          $('#epmotion-tm300-8-tool-' + plateId).val(pInfo['EpMotion TM300']);
-          $('#epmotion-tm50-8-tool-' + plateId).val(pInfo['EpMotion TM50']);
-          $('#mastermix-input-' + plateId).val(pInfo['Master mix']);
-          $('#waterlot-input-' + plateId).val(pInfo['Water lot']);
-        });
-        disableAll();
+        // Need to add a timeout so the interface is ready to set the values
+        setTimeout(function() {
```

Comment:
Is a timeout needed here or would a better way be to wait using:

```javascript
$(function() {
  // things that update the GUI
});
```

Or is the timeout waiting for something else?

Reply:
I don't think that would work. The interface that I'm waiting for here is created in the previous "addPlate" call, which executes an AJAX query, and builds the interface based on the results of the ajax query. This function already gets executed _after_ the ajax called is completed, but it is not ensured that the html that the function create is ready yet.

Reply:
I see, thanks for the info!